### ISSUE TITLE
"expand selection" behaves inconsistently

### ISSUE THREAD

**Issue by turbanisch** [microsoft/pylance-release#4393]
## Environment data

-   Language Server version: 2023.5.20
-   OS and version: Mac OS 13.3.1 (a) 
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.10.11., via Miniconda (v. 23.3.1)

## Code Snippet

![pylance](https://github.com/microsoft/pylance-release/assets/51969412/8b61213b-b57d-40bf-8765-f14221d88504)

```python
dict = {
    "key1": [1, 2],
    "key2": ["a", "two words", "the"]
}
```

## Repro Steps

1. Place cursor inside of "two" and execute `Expand Selection` multiple times.

## Expected behavior

The iterative selection should be the same each time and not be affected by a previous selection or a new line.

## Actual behavior

The first selection extends in 5 steps until the entire line is selected. After deselecting, the same selection proceeds in only 2 steps. After entering some text on a new line, the selection takes 5 steps again.

## Logs

<details>
<summary>Log</summary>

2023-05-19 16:43:14.938 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:14.939 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:17.964 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:17.966 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:18.406 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:18.406 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:18.929 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:18.930 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:19.463 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:19.463 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:19.947 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:19.951 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:21.929 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:21.930 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:23.159 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:23.160 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:23.664 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:23.664 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:25.073 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:25.074 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:25.288 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:25.288 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:25.293 [info] (62922) [FG] parsing: /Users/felix/Downloads/pandas/test.py (1ms)
2023-05-19 16:43:25.293 [info] (62922) [FG] binding: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:25.539 [info] (62922) Background analysis message: analyze
2023-05-19 16:43:25.542 [info] (62922) [BG(1)] analyzing: /Users/felix/Downloads/pandas/test.py ...
2023-05-19 16:43:25.543 [info] (62922) [BG(1)]   parsing: /Users/felix/Downloads/pandas/test.py (2ms)
2023-05-19 16:43:25.544 [info] (62922) [BG(1)]   binding: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:25.545 [info] (62922) [FG] parsing: /Users/felix/Downloads/pandas/test.py (1ms)
2023-05-19 16:43:25.545 [info] (62922) [FG] binding: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:25.546 [info] (62922) [BG(1)]   checking: /Users/felix/Downloads/pandas/test.py (3ms)
2023-05-19 16:43:25.546 [info] (62922) [BG(1)] analyzing: /Users/felix/Downloads/pandas/test.py (5ms)
2023-05-19 16:43:25.546 [info] (62922) [BG(1)] indexing: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:25.546 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:25.546 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:25.587 [info] (62922) Background analysis message: getSemanticTokens delta
2023-05-19 16:43:25.587 [info] (62922) [BG(1)] getSemanticTokens delta previousResultId:1684507341038 at /Users/felix/Downloads/pandas/test.py ...
2023-05-19 16:43:25.587 [info] (62922) [BG(1)]   parsing: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:25.588 [info] (62922) [BG(1)]   binding: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:25.588 [info] (62922) [BG(1)] getSemanticTokens delta previousResultId:1684507341038 at /Users/felix/Downloads/pandas/test.py (1ms)
2023-05-19 16:43:25.794 [info] (62922) Background analysis message: analyze
2023-05-19 16:43:25.803 [info] (62922) [BG(1)] analyzing: /Users/felix/Downloads/pandas/test.py ...
2023-05-19 16:43:25.803 [info] (62922) [BG(1)]   checking: /Users/felix/Downloads/pandas/test.py (8ms)
2023-05-19 16:43:25.804 [info] (62922) [BG(1)] analyzing: /Users/felix/Downloads/pandas/test.py (8ms)
2023-05-19 16:43:25.807 [info] (62922) Background analysis message: resumeAnalysis
2023-05-19 16:43:25.809 [info] (62922) [BG(1)] indexing: /Users/felix/Downloads/pandas/test.py [found 1] (0ms)
2023-05-19 16:43:25.810 [info] (62922) Indexing Done: /Users/felix/Downloads/pandas/test.py
2023-05-19 16:43:26.325 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:26.325 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:26.440 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:26.441 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:26.627 [info] (62922) Background analysis message: getSemanticTokens delta
2023-05-19 16:43:26.628 [info] (62922) [BG(1)] getSemanticTokens delta previousResultId:1684507405587 at /Users/felix/Downloads/pandas/test.py ...
2023-05-19 16:43:26.628 [info] (62922) [BG(1)]   parsing: /Users/felix/Downloads/pandas/test.py (1ms)
2023-05-19 16:43:26.629 [info] (62922) [BG(1)]   binding: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:26.629 [info] (62922) [BG(1)] getSemanticTokens delta previousResultId:1684507405587 at /Users/felix/Downloads/pandas/test.py (1ms)
2023-05-19 16:43:26.644 [info] (62922) [FG] parsing: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:26.644 [info] (62922) [FG] binding: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:26.683 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:26.683 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:26.739 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:26.739 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:26.767 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:26.767 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:26.860 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:26.860 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:26.979 [info] (62922) Background analysis message: getSemanticTokens delta
2023-05-19 16:43:26.980 [info] (62922) [BG(1)] getSemanticTokens delta previousResultId:1684507406628 at /Users/felix/Downloads/pandas/test.py ...
2023-05-19 16:43:26.980 [info] (62922) [BG(1)]   parsing: /Users/felix/Downloads/pandas/test.py (1ms)
2023-05-19 16:43:26.980 [info] (62922) [BG(1)]   binding: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:26.981 [info] (62922) [BG(1)] getSemanticTokens delta previousResultId:1684507406628 at /Users/felix/Downloads/pandas/test.py (1ms)
2023-05-19 16:43:27.026 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:27.027 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:27.085 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:27.085 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:27.143 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:27.143 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:27.190 [info] (62922) Background analysis message: setFileOpened
2023-05-19 16:43:27.191 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:27.327 [info] (62922) Background analysis message: getSemanticTokens delta
2023-05-19 16:43:27.327 [info] (62922) [BG(1)] getSemanticTokens delta previousResultId:1684507406980 at /Users/felix/Downloads/pandas/test.py ...
2023-05-19 16:43:27.327 [info] (62922) [BG(1)]   parsing: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:27.327 [info] (62922) [BG(1)]   binding: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:27.328 [info] (62922) [BG(1)] getSemanticTokens delta previousResultId:1684507406980 at /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:27.392 [info] (62922) [FG] parsing: /Users/felix/Downloads/pandas/test.py (1ms)
2023-05-19 16:43:27.392 [info] (62922) [FG] binding: /Users/felix/Downloads/pandas/test.py (0ms)
2023-05-19 16:43:27.442 [info] (62922) Background analysis message: analyze
2023-05-19 16:43:27.447 [info] (62922) [BG(1)] analyzing: /Users/felix/Downloads/pandas/test.py ...
2023-05-19 16:43:27.447 [info] (62922) [BG(1)]   checking: /Users/felix/Downloads/pandas/test.py (4ms)
2023-05-19 16:43:27.447 [info] (62922) [BG(1)] analyzing: /Users/felix/Downloads/pandas/test.py (4ms)
2023-05-19 16:43:27.449 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:27.449 [info] (62922) Background analysis message: resumeAnalysis
2023-05-19 16:43:27.449 [info] (62922) [BG(1)] indexing: /Users/felix/Downloads/pandas/test.py [found 1] (0ms)
2023-05-19 16:43:27.449 [info] (62922) Indexing Done: /Users/felix/Downloads/pandas/test.py
2023-05-19 16:43:27.451 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:28.489 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:28.490 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:28.674 [info] [Info  - 16:43:28] (62922) SourceFile: Received fs event 'change' for path '/Users/felix/Downloads/pandas/test.py'
2023-05-19 16:43:28.674 [info] (62922) Background analysis message: markFilesDirty
2023-05-19 16:43:28.739 [info] (62922) Background analysis message: analyze
2023-05-19 16:43:29.370 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:29.370 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:29.666 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:29.667 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:30.029 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:30.030 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:30.434 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:30.436 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:30.855 [info] (62922) Background analysis message: getDiagnosticsForRange
2023-05-19 16:43:30.856 [info] (62922) Background analysis message: getDiagnosticsForRange
</details>

**Comment by debonte** [comment]
Thanks for the repro steps @turbanisch! We were making an unintentional change to our parse tree when expand selection was run the first time. Changing the document caused a reparse which was why it worked as expected after a doc change.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.5.41, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023541-24-May-2023-PreRelease)
